src/inmanta/data/model.py

```diff
@@ -273,3 +273,36 @@ class ResourceAction(BaseModel):
     changes: Optional[JsonType]
     change: Optional[const.Change]
     send_event: Optional[bool]
+
+
+class LogLine(BaseModel):
+    class Config:
+        """
+        Pydantic config.
+        """
+
+        # Override the setting from the BaseModel class as such that the level field is
+        # serialises using the name of the enum instead of its value. This is required
+        # to make sure that data sent to the API endpoints resource_action_update
+        # and resource_deploy_done are serialized consistently using the name of the enum.
+        use_enum_values = False
+
+    @pydantic.validator("level", pre=True)
+    def level_from_enum_attribute(cls, v):
+        """
+        Ensure that a LogLine object can be instantiated using pydantic when the level field
+        is represented using the name of the enum instead of its value. This is required to
+        make sure that a serialized version of this object passes pydantic validation.
+        """
+        if isinstance(v, str):
+            try:
+                return const.LogLevel[v]
+            except KeyError:
+                raise ValueError(f"Invalid enum value {v}. Valid values: { ','.join([x.name for x in const.LogLevel]) }")
```

Comment:
```suggestion
                return v
```
I think it would be better to just leave the value unmodified if it's not a valid enum name. This way we delegate to the default pydantic validator. Suppose for example that we construct an object not through the API but from code, this would allow us to use `LogLine(level=const.LogLevel.INFO)`, which matches the field type, instead of `LogLine(level=const.LogLevel.INFO.name).

Reply:
The above-mentioned command will work due to the `isinstance()` check, but I am not opposed to checking it.

Reply:
I completely looked over the check. In that case I think it might actually be better to raise the exception as you do, so there can be no confusion on whether it accepts enum values or names.